examples/nimble_gnrc/ble.c

```diff
@@ -0,0 +1,364 @@
+    res = nimble_netif_accept(ad.buf, ad.pos, &_adv_params);
+    if (res != NIMBLE_NETIF_OK) {
+        printf("err: unable to start advertising (%i)\n", res);
+    }
+    else {
+        printf("success: advertising this node as '%s'\n", name);
+    }
+}
+
+static void _cmd_adv_stop(void)
+{
+    int res = nimble_netif_accept_stop();
+    if (res == NIMBLE_NETIF_OK) {
+        puts("canceled advertising");
+    }
+    else if (res == NIMBLE_NETIF_NOTADV) {
+        puts("no advertising in progress");
+    }
+}
+
+static void _cmd_scan(unsigned duration)
+{
+    if (duration == 0) {
+        return;
+    }
+    printf("scanning (for %ums) ... ", (duration / 1000));
+    nimble_scanlist_clear();
+    nimble_scanner_start();
+    xtimer_usleep(duration);
+    nimble_scanner_stop();
+    puts("done");
+    nimble_scanlist_print();
+}
+
+static void _cmd_connect_addr(ble_addr_t *addr)
+{
+    /* simply use NimBLEs default connection parameters */
+    int res = nimble_netif_connect(addr, NULL, APP_CONN_TIMEOUT);
+    if (res < 0) {
+        printf("err: unable to trigger connection sequence (%i)\n", res);
+        return;
+    }
+
+    printf("initiated connection procedure with ");
+    bluetil_addr_print(addr->val);
+    puts("");
+
+}
+
+static void _cmd_connect_addstr(const char *addr_str)
+{
+    /* RANDOM is the most common type, has no noticeable effect when connecting
+       anyhow... */
+    ble_addr_t addr = { .type = BLE_ADDR_RANDOM };
+
+    if (bluetil_addr_from_str(addr.val, addr_str) == NULL) {
+        puts("err: unable to parse address");
+        return;
+    }
+    _cmd_connect_addr(&addr);
+}
+
+static void _cmd_connect(unsigned pos)
+{
+    nimble_scanlist_entry_t *sle = nimble_scanlist_get_by_pos(pos);
+    if (sle == NULL) {
+        puts("err: unable to find given entry in scanlist");
+        return;
+    }
+    _cmd_connect_addr(&sle->addr);
+}
+
+static void _cmd_close(int handle)
+{
+    int res = nimble_netif_close(handle);
+    if (res != NIMBLE_NETIF_OK) {
+        puts("err: unable to close connection with given handle");
+    }
+    else {
+        puts("success: connection tear down initiated");
+    }
+}
+
+static void _cmd_update(int handle, int itvl, int timeout)
+{
+    struct ble_gap_upd_params params;
+    params.itvl_min = (uint16_t)((itvl * 1000) / BLE_HCI_CONN_ITVL);
+    params.itvl_max = (uint16_t)((itvl * 1000) / BLE_HCI_CONN_ITVL);
+    params.latency = 0;
+    params.supervision_timeout = (uint16_t)(timeout / 10);
+    params.min_ce_len = BLE_GAP_INITIAL_CONN_MIN_CE_LEN;
+    params.max_ce_len = BLE_GAP_INITIAL_CONN_MAX_CE_LEN;
+
+    int res = nimble_netif_update(handle, &params);
+    if (res != NIMBLE_NETIF_OK) {
+        puts("err: unable to update connection parameters for given handle");
+    }
+    else {
+        puts("success: connection parameters updated");
+    }
+}
+
+static int _ishelp(char *argv)
+{
+    return memcmp(argv, "help", 4) == 0;
+}
+
+void app_ble_init(void)
```

Comment:
Couln't this and the callback go to `netif->ops->init()`? It does not look very dependent on the application for me.

Reply:
but it is.

The `scanlist´ has nothing to do with `nimble/netif`, and the callback is only valid for this application...